packages/swap/src/quoting/Quoter.ts

```diff
@@ -97,11 +120,45 @@ export default class Quoter {
     });
   }
 
-  private async collectMakerQuotes(request: MarketMakerQuoteRequest): Promise<MarketMakerQuote[]> {
+  private async collectMakerQuotes(
+    request: MarketMakerQuoteRequest<Leg>,
+  ): Promise<MarketMakerQuote[]> {
     const connectedClients = this.io.sockets.sockets.size;
     if (connectedClients === 0) return Promise.resolve([]);
 
-    this.io.emit('quote_request', request);
+    let expectedResponses = 0;
+
+    const quotedLegsMap = new Map<string, LegFormatter>();
+
+    for (const socket of await this.io.fetchSockets()) {
+      const [first, second] = request.legs;
+      const { quotedAssets, marketMaker } = socket.data;
+      const quotesFirstLeg = quotedAssets[first.getBaseAsset()];
+      const quotesSecondLeg = !second || quotedAssets[second.getBaseAsset()];
```

Comment:
i dont understand the `!second` here but im sure there is a reason

Reply:
If there is no second leg

Reply:
but why do we need to set `quotesSecondLeg` to true then?

Reply:
So the first `if` statement below is used. Can probably simplify/make this more explicit